Objects/regions.c

```diff
@@ -88,6 +89,168 @@ bool is_c_wrapper(PyObject* obj){
     return PyCFunction_Check(obj) || Py_IS_TYPE(obj, &_PyMethodWrapper_Type) || Py_IS_TYPE(obj, &PyWrapperDescr_Type);
 }
 
+/**
+ * Global status for performing the region check.
+ */
+bool do_region_check = false;
+
+// The src object for an edge that invalidated the invariant.
+PyObject* error_src = Py_None;
+
+// The tgt object for an edge that invalidated the invariant.
+PyObject* error_tgt = Py_None;
+
+// Once an error has occurred this is used to surpress further checking
+bool error_occurred = false;
+
+
+/**
+ * Enable the region check.
+ */
+void notify_regions_in_use(void)
+{
+    // Do not re-enable, if we have detected a fault.
+    if (!error_occurred)
+        do_region_check = true;
+}
+
+PyObject* _Py_EnableInvariant(void)
+{
+    // Disable failure as program has explicitly requested invariant to be checked again.
+    error_occurred = false;
+    // Re-enable region check
+    do_region_check = true;
+    return Py_None;
+}
+
+/**
+ * Set the global variables for a failure.
+ * This allows the interpreter to inspect what has failed.
+ */
+void set_failed_edge(PyObject* src, PyObject* tgt)
+{
+    Py_IncRef(src);
+    error_src = src;
+    Py_IncRef(tgt);
+    error_tgt = tgt;
+    printf("Error: %p -> %p: destination is not immutable\n", src, tgt);
```

Comment:
NIT: I'd suggest making this message more generic:

```suggestion
    printf("Error: %p -> %p: invalid pointer found from %p -> %p\n", src, tgt);
```

with the newly added `%p` for the region pointers.

Reply:
> NIT: I'd suggest making this message more generic:

@xFrednet Why?

Reply:
Because the function name `set_failed_edge` implies that it will be called for all kinds of edge failures, not just immutable edges. For now, this message is correct, but it'll quickly become outdated. Since immutable are a constant (-1 I believe) it would still be easy to see.

Reply:
Got it. I suggest we wait with that until the commit where we add more edge failures!